### ISSUE TITLE
Can't run app from background after notification tap

### ISSUE THREAD

**Issue by kessero** [OneSignal/OneSignal-Flutter-SDK#233]
**Description:**

I get notifications but after clicking  the notification, when the application is in the background or turned off it does not launch the application. Log indicates that setNotificationOpenedHandler is launched after clicking.
In LOG:
D/FlutterView( 8279): Detaching from a FlutterEngine: io.flutter.embedding.engine.FlutterEngine@def89a3
E/ShortcutBadger( 8279): Unable to find launch intent for package com.example.demo
**E/ShortcutBadger( 8279): Unable to find launch intent for package com.example.demo**??
D/onesignal( 8279): Created json raw payload: ...
E/OneSignal( 8279): Error updating the user record because of th enull user id
E/ShortcutBadger( 8279): Unable to find launch intent for package com.example.demo
D/onesignal( 8279): Created json raw payload: {vis=1, google.original_priority=high, google.sent_time=1586967131840,...}
I/flutter ( 8279): Payload only is{
I/flutter ( 8279):   "google.delivered_priority": "high",
I/flutter ( 8279):   "google.sent_time": 1586967131840,
I/flutter ( 8279):   "google.ttl": 259200,
I/flutter ( 8279):   "google.original_priority": "high",...}

**Environment**
onesignal_flutter: ^2.4.0

**Steps to Reproduce Issue:**

1. Install the OneSignal Flutter SDK using pub into your project
My code:
`
Future<void> configOneSignal() async {
        if (!mounted) return;    
          var settings = {
            OSiOSSettings.autoPrompt: false,
            OSiOSSettings.promptBeforeOpeningPushUrl: true            
          };
          /// The notification received handler will be called whenever a notification
          /// is received by the SDK (only applies to OneSignal push notifications)
          OneSignal.shared.setNotificationReceivedHandler((notification) {            
          });                   
          // The notification opened handler is called whenever the user opens a
          /// OneSignal push notification, or taps an action button on a notification.
          OneSignal.shared.setNotificationOpenedHandler((OSNotificationOpenedResult result) {

            print("Payload only is"+result.notification.payload.jsonRepresentation().toString());
            Navigator.push(
              context,
              MaterialPageRoute(
                builder: (context) => HomePage(),
              ),
            );                 
          });
          
          OneSignal.shared.setSubscriptionObserver((OSSubscriptionStateChanges changes) {
            print("SUBSCRIPTION STATE CHANGED: ${changes.jsonRepresentation()}");
          });         

          OneSignal.shared.setPermissionObserver((OSPermissionStateChanges changes) {
            print("PERMISSION STATE CHANGED: ${changes.jsonRepresentation()}");
          });

        
          // NOTE: Replace with your own app ID from https://www.onesignal.com
          await OneSignal.shared.init("id", iOSSettings: settings);
        
          OneSignal.shared.consentGranted(true);
          OneSignal.shared.setInFocusDisplayType(OSNotificationDisplayType.notification);
          await OneSignal.shared.getPermissionSubscriptionState();
          bool requiresConsent = await OneSignal.shared.requiresUserPrivacyConsent(); 
                 
        }   `

**Anything else:**

(crash stacktraces, as well as any other information here)

**Comment by rgomezp** [comment]
Howdy,
It looks like user-id is `null`. You should always initialize before anything else.

[Src](https://github.com/OneSignal/OneSignal-Android-SDK/blob/master/OneSignalSDK/onesignal/src/main/java/com/onesignal/UserStateSynchronizer.java#L309)

**Comment by vernalclareai** [comment]
I also got the same issue.

**Comment by vernalclareai** [comment]
How can I generate the user-id for initializing of the SDK?
Thanks.

**Comment by rgomezp** [comment]
Duplicate of https://github.com/OneSignal/OneSignal-Android-SDK/issues/1008

Please continue tracking issue progress in the Android repository as this pertains to Android in general

**Comment by rgomezp** [comment]
@vernalclareai 
The user-id is generated only after a device has subscribed to push with OneSignal. However, if you would like, you can generate your own device identifier beforehand and then assign it to the OneSignal "player" (user) record with `setExternalUserId` ([reference](https://documentation.onesignal.com/docs/sdk-reference#setexternaluserid-method))

**Comment by harpreet9629** [comment]
@kessero in your app Navigator.push working when open notification?
